### ISSUE TITLE
FM几个改进建议及问题

### ISSUE THREAD

**Issue by JWenBin** [shenweichen/DeepMatch#60]
建议：
1. FM里面写死了binary，没提供regression，我试过在我的千万级数据量的时候回归拟合rating分数的效果比binary分类得到的auc等各项指标要好；然后loss对应用的"mse"。

问题：
1. fm.py中没看见一阶线性部分、二价交叉部分模型结构，封装的部分还没来得及仔细看，，，，
2. 改了下fm.py成：score = tf.reduce_sum(Add()([user_vector_sum, item_vector_sum]), axis=1, keepdims=True)，收敛很慢，auc也掉到了0.66远不如原始的余弦相似度，也不如矩阵分解。
3. 召回率居然不及矩阵分解，虽然测试集auc指标能到0.92(训练集0.96)，但是负样本约0.5分，正样本约0.85分，召回数据top500都是0.99分的，以至于离线评估召回率比不过矩阵分解。

**Comment by TzeSing** [comment]
我看了下，没有一阶和二阶的操作，就只是把user_id_emb和其他user_side_info_emb sum成一个emb，同理item侧也是，最后走了个cos，这好像不能叫FM吧
